exporter/metric/metric.go

```diff
@@ -272,25 +291,92 @@ func (me *metricExporter) recordToMdpb(record *export.Record) *googlemetricpb.Me
 	}
 }
 
+// checkPrefixInMapKeys checks whether the non-empty prefix string
+//  is in the values of the map
+func checkPrefixInMapKeys(labelMap map[string]string, prefix string) bool {
+	if len(prefix) == 0 {
+		return false
+	}
+	for key := range labelMap {
+		if strings.HasPrefix(key, prefix) {
+			return true
+		}
+	}
+	return false
+}
+
 // resourceToMonitoredResourcepb converts resource in OTel to MonitoredResource
 // proto type for Cloud Monitoring.
 //
 // https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.monitoredResourceDescriptors
-func (me *metricExporter) resourceToMonitoredResourcepb(_ *resource.Resource) *monitoredrespb.MonitoredResource {
-	// TODO: Implement the process to convert Resource to MonitoringResoruce.
-	// ref. https://cloud.google.com/monitoring/api/resources
-	// This method only returns global resource for now.
-
-	// "global" only accepts "project_id" for label.
-	// https://cloud.google.com/monitoring/api/resources#tag_global
+func (me *metricExporter) resourceToMonitoredResourcepb(res *resource.Resource) *monitoredrespb.MonitoredResource {
+
+	if res == nil || res.Len() == 0 {
+    	// Return "global" Monitored resources if the input resource is null or empty
+	    // "global" only accepts "project_id" for label.
+	    // https://cloud.google.com/monitoring/api/resources#tag_global
+		return &monitoredrespb.MonitoredResource{
+			Type: "global",
+			Labels: map[string]string{
+				"project_id": me.o.ProjectID,
+			},
+		}	
+	}
+
+	// labels in Resource are in the form of []kv.KeyValue
+	// convert them into a map of kv.String
+	resLabelList := res.Attributes()
+
+	resLabelMap := make(map[string]string)
```

Comment:
I'm okay if you leave this but in terms of performance, note that it's probably not worth building and allocating this working map vs just iterating through the collection 1-2 extra times. In fact I think with the current code it'd be the same number of iterations.

i.e.
Currently you iterate:
- Once to generate the map
- Once to determine if a "k8s" key exists

Instead you could iterate:
- Once to find CloudKeyProvider
- Once to determine if "k8s" key exists
(then iterate through `input` instead of `match` in `transformResource`)

Having said that, it might be easier to maintain in the future with a map as we add more resource types, and we can benchmark / fine tune this in the future.

Instead, consider at least putting the code to build this map into a separate function to improve readability.